torchchat/generate.py

```diff
@@ -591,6 +591,7 @@ def generate(
             Dict[str, Any]
         ] = None,  # List of Image prompt tensors for multimodal models
         start_pos: int = 0,
+        skip_cache_setup: bool = False,
```

Comment:
I'm ok for now, but introducing new inputs into generate function might trigger my nightmare 😣, making it farther away from our target.

Reply:
I agree, it's not great. Luckily it's light so we can abstract it easily later on